src/extension.ts

```diff
@@ -133,3 +152,7 @@ export function noWorkspaceFolder(uri: Uri): WorkspaceFolder {
 function isValidCrystalDocument(doc: TextDocument): boolean {
     return doc.languageId === 'crystal' && !doc.isUntitled && doc.uri.scheme === 'file'
 }
```

Comment:
With `isDocumentVirtual` function we can make this one more more constrained.

```suggestion
function isCrystalDocument(doc: TextDocument): boolean {
    return doc.languageId === 'crystal'
}
```